target_chains/sui/contracts/sources/pyth.move

```diff
@@ -651,12 +776,329 @@ module pyth::pyth_tests{
+        assert!(vector::length<ID>(&created_ids)==4, 0);
+
+        let price_info_object_1 = take_shared<PriceInfoObject>(&scenario);
+        let price_info_object_2 = take_shared<PriceInfoObject>(&scenario);
+        let price_info_object_3 = take_shared<PriceInfoObject>(&scenario);
+        let price_info_object_4 = take_shared<PriceInfoObject>(&scenario);
+
+        // Create vector of price info objects (Sui objects with key ability and living in global store),
+        // which contain the price feeds we want to update. Note that these can be passed into
+        // update_price_feeds in any order!
+        //let price_info_object_vec = vector[price_info_object_1, price_info_object_2, price_info_object_3, price_info_object_4];
+        verified_vaas = get_verified_test_vaas(&worm_state, &clock);
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+
+        let vaa_1 = vector::pop_back<VAA>(&mut verified_vaas);
+
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+
+        // Create authenticated price infos
+        let vec = create_authenticated_price_infos_using_batch_price_attestation(
+            &pyth_state,
+            vector[vaa_1],
+            &clock
+        );
+
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+        vec = update_single_price_feed(
+            &mut pyth_state,
+            vec,
+            &mut price_info_object_1,
+            test_coins,
+            &clock
+        );
+
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+        authenticated_vector::destroy<PriceInfo>(vec);
+
+        vector::destroy_empty(verified_vaas);
+        return_shared(pyth_state);
+        return_shared(worm_state);
+        return_shared(price_info_object_1);
+        return_shared(price_info_object_2);
+        return_shared(price_info_object_3);
+        return_shared(price_info_object_4);
+
+        clock::destroy_for_testing(clock);
+        test_scenario::end(scenario);
+    }
+
+
+    #[test_only]
+    const TEST_ACCUMULATOR_SINGLE_FEED: vector<u8> = x"504e41550100000000a0010000000001005d461ac1dfffa8451edda17e4b28a46c8ae912422b2dc0cb7732828c497778ea27147fb95b4d250651931845e7f3e22c46326716bcf82be2874a9c9ab94b6e42000000000000000000000171f8dcb863d176e2c420ad6610cf687359612b6fb392e0642b0ca6b1f186aa3b0000000000000000004155575600000000000000000000000000da936d73429246d131873a0bab90ad7b416510be01005500b10e2d527612073b26eecdfd717e6a320cf44b4afac2b0732d9fcbe2b7fa0cf65f958f4883f9d2a8b5b1008d1fa01db95cf4a8c7000000006491cc757be59f3f377c0d3f423a695e81ad1eb504f8554c3620c3fd02f2ee15ea639b73fa3db9b34a245bdfa015c260c5a8a1180177cf30b2c0bebbb1adfe8f7985d051d2";
+    #[test]
+    fun test_create_and_update_single_price_feed_with_accumulator_success() {
+        use pyth::data_source::Self;
+
+        let base_update_fee = 50;
+        let coins_to_mint = 5000;
+        let emitter_address = x"71f8dcb863d176e2c420ad6610cf687359612b6fb392e0642b0ca6b1f186aa3b";
```

Comment:
suggest extracting some constants for these byte strings so they can be reused across tests instead of defined repeatedly.

Reply:
i also suggest leaving a comment explaining what's in the accumulator payload above, because I can't really tell what the test does otherwise.

Reply:
i notice that test cases generally have quite a bit of repeated code. Is there any way to reduce that? It's a lot easier for the reader to understand what's important (or different) about each test that way.

Maybe you could make helpers like `test_accumulator_update_is_successful(message, expected_price_infos)` and `test_accumulator_update_fails(message, error code)` or something like that ?

Reply:
this specific thing is kind of hard to do in Move, since for the failure case, the entire #[test] function has to be annotated with the error code at the top, e.g.

```
#[test]
#[expected_failure(abort_code = pyth::accumulator::E_INVALID_PROOF)]
    fun test_create_and_update_single_price_feed_with_accumulator_failure() {
```
so passing in the error code doesn't help.

I'll try to reduce the amount of shared code between functions as much as possible by factoring out constants and hard-coding them, writing helper functions etc.